Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs

```diff
@@ -0,0 +1,143 @@
+        private Timer _animationTimer;
+        #endregion
+        #region Binding Properties
+        public static readonly BindableProperty SpinnerTypeProperty =
+            BindableProperty.Create(nameof(SpinnerType), typeof(SpinnerType), typeof(TMSpinner), defaultValue: SpinnerType.InDeterminate, propertyChanged: OnSpinnerChanged);
+
+        public static readonly BindableProperty SpinnerColorProperty =
+            BindableProperty.Create(nameof(SpinnerColor), typeof(SpinnerColor), typeof(TMSpinner), defaultValue: SpinnerColor.Blue, propertyChanged: OnSpinnerColorChanged);
+        #endregion
+        #region Public Properties
+        public SpinnerType SpinnerType
+        {
+            get => (SpinnerType)GetValue(SpinnerTypeProperty);
+            set => SetValue(SpinnerTypeProperty, value);
+        }
+        public SpinnerColor SpinnerColor
+        {
+            get => (SpinnerColor)GetValue(SpinnerColorProperty);
+            set => SetValue(SpinnerColorProperty, value);
+        }
+        #endregion
+        public TMSpinner()
+        {
+            WidthRequest = minWidth;
+            HeightRequest = minHeight;
+            IgnorePixelScaling = false;
+            StartAnimation();
+        }
+        #region Protected Methods
+        protected override void OnPaintSurface(SKPaintSurfaceEventArgs e)
+        {
+            base.OnPaintSurface(e);
+
+            DrawCircle(e);
+        }
+        #endregion
+        #region Private Methods
+        private static void OnSpinnerColorChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            if (bindable is TMSpinner tmSpinner)
+            {
+                tmSpinner._spinnerColor = ((SpinnerColor)newValue == SpinnerColor.White) ? ResourcesDictionary.ColorsDictionary(ColorsConstants.White)
+                    : ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+            }
+        }
+        private static void OnSpinnerChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            if (bindable is TMSpinner tmSpinner)
+            {
+                tmSpinner._spinnerType = (SpinnerType)newValue;
+            }
+        }
+        private void StartAnimation()
+        {
+            _animationTimer = new Timer(UpdateAnimation, null, TimeSpan.Zero, TimeSpan.FromMilliseconds(animateTimerSeconds));
+        }
+
+        private void UpdateAnimation(object state)
+        {
+            if (_spinnerType == SpinnerType.InDeterminate)
+            {
+                _rotationAngle += 5f;
+            }
+            else if (_spinnerType == SpinnerType.Determinate)
+            {
+                _startAngle += 5f;
+                _sweepAngle += 5f;
+                if (_sweepAngle >= 360f)
+                {
+                    _sweepAngle = 0f;
+                    _startAngle = 0f;
+                }
+            }
+            Device.BeginInvokeOnMainThread(() =>
```

Comment:
I don't think we need the `Device.BeginInvokeOnMainThread( ` here, please check once. Even if we need, Device.BeginInvokeOnMainThread is deprecated now, we shouldn't use it.

Reply:
We need it hence changed to MainThread.BeginInvokeOnMainThread